src/components/Main/Main.tsx

```diff
@@ -0,0 +1,51 @@
+import React, { useContext, useState } from 'react';
+import { TodoItem } from '../TodoItem/TodoItem';
+
+import { DispatchContext, StateContext } from '../../Store';
+import { Status } from '../../types/Status';
+import { Todo } from '../../types/Todo';
+
+export const Main: React.FC = () => {
+  const dispatch = useContext(DispatchContext);
+  const { todos, filter } = useContext(StateContext);
+
+  const [toggle, setToggle] = useState(false);
+
+  const handleOnToggle = () => {
+    dispatch({ type: 'toggleAll', value: !toggle });
+    setToggle(!toggle);
+  };
+
+  const preparedTodos = [...todos].filter(todo => {
```

Comment:
```suggestion
  const preparedTodos = todos.filter(todo => {
```
filter doesn't mutate the array